kyo-scheduler/jvm/src/main/scala/kyo/scheduler/InternalClock.scala

```diff
@@ -32,6 +32,8 @@ final private[kyo] case class InternalClock(executor: Executor) {
     def stop(): Unit =
         _stop = true
 
-    private val gauge =
-        statsScope.scope("clock").gauge("skew")((System.currentTimeMillis() - millis).toDouble)
+    locally {
+        val _ =
+            statsScope.scope("clock").gauge("skew")((System.currentTimeMillis() - millis).toDouble)
```

Comment:
Metrics are held via weak references in `kyo-stats-registry` so we need the class field holding the `gauge` reference to avoid it getting garbage collected. Maybe we could move [`kyo.discard`](https://github.com/getkyo/kyo/blob/6e152e77f780fe5e9af5c03a1716b878d9229666/kyo-core/shared/src/main/scala/kyo/package.scala#L62) to `kyo-scheduler`?